dapr_agents/llm/dapr/utils.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import time
+from typing import Any, Dict
+
+from dapr_agents.types.message import (
+    AssistantMessage,
+    LLMChatCandidate,
+    LLMChatResponse,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def process_dapr_chat_response(response: Dict[str, Any]) -> LLMChatResponse:
+    """
+    Convert a Dapr-normalized chat dict (with OpenAI-style 'choices') into a unified LLMChatResponse.
+
+    Args:
+        response: The dict returned by `DaprChatClient.translate_response`.
+
+    Returns:
+        LLMChatResponse: Contains a list of candidates and metadata.
+    """
+    # 1) Extract each choice → build AssistantMessage + LLMChatCandidate
+    candidates = []
+    for choice in response.get("choices", []):
+        msg = choice.get("message", {})
+        assistant_message = AssistantMessage(
+            content=msg.get("content"),
+            # Dapr currently never returns refusals, tool_calls or function_call here
```

Comment:
I'll update here when we bump dapr runtime versions bc now we will return tool_calls. Function_calls are deprecated from what I saw, so we will never return that. Also, under the hood in dapr runtime we use LangChain Go SDK which actually currently does not support refusal msg, so we wont get that either until langchain uses it :)